### PR TITLE
fix: don't replicate local setup

### PR DESCRIPTION
**Pull request by ioggstream**
The following patch:
 - avoid replicating local user setup to slave hosts;
 - pass --defaults-file to mysql_install_db

### CODE REVIEW

File: 5.6/docker-entrypoint.sh [hunk 1]
```diff
@@ -26,6 +29,10 @@ if [ "$1" = 'mysqld' ]; then
 		
 		tempSqlFile='/tmp/mysql-first-time.sql'
 		cat > "$tempSqlFile" <<-EOSQL
+		        -- What's done in this file shouldn't be replicated
```
Comment (tianon) [inline 5.6/docker-entrypoint.sh]: Mixed tabs and spaces here. :wink:

File: 5.6/docker-entrypoint.sh [hunk 2]
```diff
@@ -17,7 +18,9 @@ if [ "$1" = 'mysqld' ]; then
 		fi
 		
 		echo 'Running mysql_install_db ...'
-		mysql_install_db --datadir="$DATADIR"
+		# pass --defaults-file and --defaults-extra-file to mysql_install_db
+		DEFAULTS_FILE_ARGS=$(egrep -o -- '--defaults(|-extra)-file=[^ ]+' <<< "$@")
+		mysql_install_db --datadir="$DATADIR" $DEFAULTS_FILE_ARGS
```
Comment (tianon) [inline 5.6/docker-entrypoint.sh]: This change is a lot more fragile (and thus controversial) than the `SET @@SESSION.SQL_LOG_BIN` is.

Comment (ioggstream) [inline 5.6/docker-entrypoint.sh]: Thx @tianon for your review!

You don't like the code or the functionality?

File: 5.6/docker-entrypoint.sh (hunk 1 above)
Comment (ioggstream) [inline 5.6/docker-entrypoint.sh]: fixing...(I got kate in python mode)

**Comment by yosifkit** [comment]
LGTM

**Comment by tianon** [comment]
LGTM

**Pull request merged**